src/BODSDataExtractor/extractor.py

```diff
@@ -68,22 +68,49 @@ def __init__(self, api_key, limit=10000, nocs=None, status='published', search=N
             self.generate_timetable()
 
         # self.service_line_extract = service_line_extract
+     
+    def check_api_response(self, response):
+
+        #initialise empty message to be appended to
+        message=""
+
+
+        if response.get("results")==[]:
+            response={'status_code': 400, 'reason': '{"Invalid Entry in data-object"}'}

```

Comment:
Didn't I see code before where you were actually pulling the status code from the response object? Has that been removed?

Reply:
The code where we get the status code from the response object has been removed. This is because we can only successfully access the response object after executing all lines of code. In that case the status code and reason were taken at the end of executing all lines of code, so it gave us a status code of 200 and an OK message.